src/MessagePack.UnityClient/Assets/MessagePack.UnityClient.Dependencies.msb4u.csproj

```diff
@@ -0,0 +1,42 @@
+<Project ToolsVersion="15.0">
+  <!--GENERATED FILE-->
+  <!--
+    This file can be modified and checked in.
+
+    It is different from the other generated C# Projects in that it will be the one gathering all dependencies and placing them into the Unity asset folder.
+
+    You can add project level dependencies to this file, by placing them below:
+    - <Import Project="$(MSBuildForUnityGeneratedProjectDirectory)\$(MSBuildProjectName).g.props" />
+    and before:
+    - <Import Project="$(MSBuildForUnityGeneratedProjectDirectory)\$(MSBuildProjectName).g.targets" />
+
+    Do not add any source or compilation items.
+
+    Examples of how you can modify this file:
+    - Add NuGet package references:
+        <ItemGroup>
+          <PackageReference Include="Newtonsoft.Json" Version="11.0.1" />
+        </ItemGroup>
+    - Add external C# project references:
+      <ItemGroup>
+        <ProjectReference Include="..\..\..\ExternalLib\ExternalLib.csproj" />
+      </ItemGroup>
+  -->
+
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(MSBuildForUnity.Common.props))" Condition="Exists('$([MSBuild]::GetPathOfFileAbove(MSBuildForUnity.Common.props))')" />
+
+  <PropertyGroup>
+    <UnityCurrentTargetFramework>net471</UnityCurrentTargetFramework>
+    <TargetFramework>net471</TargetFramework>
+  </PropertyGroup>
+
+  <!-- SDK.props is imported inside this props file -->
+  <Import Project="$(MSBuildForUnityGeneratedProjectDirectory)\$(MSBuildProjectName).g.props" />
+
+  <ItemGroup>
+    <!--Add NuGet or Project references here-->
```

Comment:
What kind of project references work here? Could we add a project or package reference for messagepack now instead of placing all the source directly into the unity folder as we do today?

Reply:
You can add nuget packages or project references for net471 or netstandard2.0.
If you add a reference here, the dll file will be copied to `Assets/Dependencies/`.
But the references needed for `Assets/Scripts/MessagePack/` must be added to `Assets/Scripts/MessagePack/MessagePack.csproj`.

https://github.com/microsoft/MSBuildForUnity/blob/master/Samples/SimpleNuGetDependency.Unity/README.md